vslib/src/VirtualSwitchSaiInterface.cpp

```diff
@@ -789,6 +789,37 @@ sai_status_t VirtualSwitchSaiInterface::objectTypeGetAvailability(
     return SAI_STATUS_NOT_SUPPORTED;
 }
 
+sai_status_t VirtualSwitchSaiInterface::queryAattributeCapability(
+        _In_ sai_object_id_t switch_id,
+        _In_ sai_object_type_t object_type,
+        _In_ sai_attr_id_t attr_id,
+        _Out_ sai_attr_capability_t *capability)
+{
+    SWSS_LOG_ENTER();
+
+    // TODO: We should generate this metadata for the virtual switch rather
+    // than hard-coding it here.
+
+    if (object_type == SAI_OBJECT_TYPE_PORT && attr_id == SAI_PORT_ATTR_TPID)
```

Comment:
in virtual switch by default all apis are implemented for all objects, so you could just set true for all without if's

Reply:
Added the comment as you suggested and removed the checking to always return success regardless of the Attribute involved and set them all implemented.